### PR TITLE
POC: LSTM net and new methods for tsdataset

### PR DESCRIPTION
**Pull request by martins0n**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: etna/datasets/tsdataset.py [hunk 1]
```diff
@@ -1122,3 +1126,76 @@ def info(self, segments: Optional[Sequence[str]] = None) -> None:
...
+
+        df = self.make_future(decoder_length, encoder_length + 1).to_pandas(flatten=True)
+
+        ts_segments = [df_segment for _, df_segment in df.groupby("segment")]
+        ts_samples = [
+            i
+            for dict_segment in ts_segments
+            for i in make_samples(dict_segment, encoder_length, decoder_length, columns_to_add)
+            if not i["encoder_real"].isnan().any()
+        ]
```
Comment (iKintosh) [inline etna/datasets/tsdataset.py:1156]: This is a full duplicate of https://github.com/tinkoff-ai/etna/blob/bbf6f9603c406379a582d6c9f940899535c8979e/etna/datasets/tsdataset.py#L1135-L1141

File: etna/datasets/tsdataset.py [hunk 2]
```diff
@@ -1122,3 +1126,76 @@ def info(self, segments: Optional[Sequence[str]] = None) -> None:
         # print the results
         result_string = "\n".join(lines)
         print(result_string)
+
+    def to_train_dataloader(self, encoder_length, decoder_length, columns_to_add, batch_size: int = 1):
```
Comment (iKintosh) [inline etna/datasets/tsdataset.py:1130]: Typings are necessary.

File: etna/datasets/tsdataset.py [hunk 3]
```diff
@@ -1122,3 +1126,76 @@ def info(self, segments: Optional[Sequence[str]] = None) -> None:
...
+            i
+            for dict_segment in ts_segments
+            for i in make_samples(dict_segment, encoder_length, decoder_length, columns_to_add)
+            if not i["encoder_real"].isnan().any()
+        ]
+
+        return DataLoader(ts_samples, batch_size=batch_size)
+
+
+def make_samples(x: dict, encoder_length, decoder_length, columns_to_add):
```
Comment (iKintosh) [inline etna/datasets/tsdataset.py:1161]: Also typings

File: etna/datasets/tsdataset.py [hunk 4]
```diff
@@ -1122,3 +1126,76 @@ def info(self, segments: Optional[Sequence[str]] = None) -> None:
...
+        ts_samples = [
+            i
+            for dict_segment in ts_segments
+            for i in make_samples(dict_segment, encoder_length, decoder_length, columns_to_add)
+            if not i["encoder_real"].isnan().any()
+        ]
+
+        return DataLoader(ts_samples, batch_size=batch_size)
+
+    def to_test_dataloader(self, encoder_length, decoder_length, columns_to_add, batch_size: int = 1):
```
Comment (iKintosh) [inline etna/datasets/tsdataset.py:1145]: Also typings. They are needed not only for parameters but for output of the methods as well.

File: etna/models/nn/utils.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from typing import TypedDict
+
+import torch
+
+
+class TrainBatch(TypedDict):
+    encoder_real: torch.Tensor
+    decoder_real: torch.Tensor
+    target: torch.Tensor
```
Comment (iKintosh) [inline etna/models/nn/utils.py:9]: Why not use dataclass? 
https://docs.python.org/3/library/dataclasses.html

**Review by iKintosh — CHANGES_REQUESTED** [review]
Everything looks good. But need to add typings and fix style problems.

**Comment by martins0n** [comment]
closed in #776

**Pull request closed without merging**